irksome/nystrom_stepper.py

```diff
@@ -0,0 +1,190 @@
+from .base_time_stepper import StageCoupledTimeStepper
+from .bcs import BCStageData, bc2space
+from .deriv import Dt, TimeDerivative, expand_time_derivatives
+from .tools import component_replace, replace, vecconst
+from firedrake import TestFunction, as_ufl
+import numpy
+from ufl import zero
+
+
+class NystromTableau:
+    def __init__(self, A, b, c, Abar, bbar, order):
+        self.A = A
+        self.b = b
+        self.c = c
+        self.Abar = Abar
+        self.bbar = bbar
+        self.order = order
+
+    @property
+    def num_stages(self):
+        return len(self.b)
+
+    @property
+    def is_explicit(self):
+        return (numpy.allclose(self.Abar, 0)
```

Comment:
Should be `numpy.allclose(numpy.triu(self.Abar), 0)` here, too

Reply:
Got it